### PR TITLE
Add xtensor-python recipe

### PR DESCRIPTION
**Pull request by SylvainCorlay**
The fixed requirements on xtensor and pybind are temporary. Next version will have looser requirements.

### CODE REVIEW

File: recipes/xtensor-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+test:
+  commands:
+    - test -d ${PREFIX}/include/xtensor-python  # [unix]
+    - test -f ${PREFIX}/include/xtensor-python/pyarray.hpp  # [unix]
+    - if exist %PREFIX%\include\xtensor-python\pyarray.hpp (exit 0) else (exit 1)  # [win]
+
+about:
+  home: http://quantstack.net/xtensor
+  license: BSD
```
Comment (jakirkham) [inline recipes/xtensor-python/meta.yaml]: Also can optionally add `license_family: BSD`.

File: recipes/xtensor-python/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  license_file: LICENSE
+  summary: 'Python bindings for xtensor, the C++ tensor algebra library'
+  description: 'Multi dimensional arrays with broadcasting and lazy computing'
+  doc_url: http://xtensor.readthedocs.io
+  dev_url: https://github.com/QuantStack/xtensor-python
+
+extra:
+  recipe-maintainers:
+    - SylvainCorlay
+    - JohanMabille
```
Comment (jakirkham) [inline recipes/xtensor-python/meta.yaml]: Please confirm that it is ok for you to be a maintainer on this, @JohanMabille .

File: recipes/xtensor-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    url: https://github.com/QuantStack/xtensor-python/archive/{{ version }}.tar.gz
+    sha256: {{ sha256 }}
+
+build:
+  number: 0
+
+requirements:
+  run:
+    - xtensor ==0.1.1
+    - pybind11 ==1.8.1
```
Comment (SylvainCorlay) [inline recipes/xtensor-python/meta.yaml]: technically it is a header-only library...

File: recipes/xtensor-python/meta.yaml (hunk 2 above)
Comment (JohanMabille) [inline recipes/xtensor-python/meta.yaml]: yep, no problem

**Comment by SylvainCorlay** [comment]
Merge?

**Comment by SylvainCorlay** [comment]
@jakirkham merge?

**Pull request merged**